src/app/account/account-container/account-details.container.ts

```diff
@@ -96,9 +95,28 @@ export class AccountDetailsContainerComponent extends WithUnsubscribe() implemen
       });
   }
 
-
   public isAdmin() {
     return this.authService.isAdmin();
   }
 
+  private setNoLimitToInfinity(limit: ResourceLimit) {
+    if (limit.max !== -1) {
+      return limit;
+    }
+    return {
+      ...limit,
+      max: Infinity
+    };
+  }
+
+  private setInfinityToNoLimit(limit: ResourceLimit) {
+    const max: number | string = limit.max as number | string;
```

Comment:
this is strange.
ResourceLimit is an interface with max as a number. You need to follow it. otherwise, do not specify that kind of an interface. Try to pass ResourceLimit[] to onLimitsUpdate